src/event/event.rs

```diff
@@ -62,59 +62,51 @@ impl Event {
         sys::event::is_error(&self.inner)
     }
 
-    /// Returns true if the event contains HUP readiness.
+    /// Returns true if the event contains read close readiness.
     ///
     /// # Notes
     ///
-    /// Method is available on all platforms, but not all platforms trigger the
-    /// HUP event.
-    ///
-    /// Because of the above be cautions when using this in cross-platform
-    /// applications, Mio makes no attempt at normalising this indicator and
-    /// only provides a convenience method to read it. Refer to the selector
-    /// documentation (below) when using this indicator.
+    /// Method is a best effort implementation. While some platforms may not
+    /// return readiness when read half is closed, it is guaranteed that
+    /// false-positives will not occur.
     ///
     /// The table below shows what flags are checked on what OS.
     ///
     /// | [OS selector] | Flag(s) checked |
     /// |---------------|-----------------|
-    /// | [epoll]       | `EPOLLHUP`      |
+    /// | [epoll]       | `EPOLLHUP`
+    /// |               | `EPOLLIN` and `EPOLLRDHUP` |
     /// | [kqueue]      | `EV_EOF`        |
     ///
     /// [OS selector]: ../struct.Poll.html#implementation-notes
     /// [epoll]: http://man7.org/linux/man-pages/man7/epoll.7.html
     /// [kqueue]: https://www.freebsd.org/cgi/man.cgi?query=kqueue&sektion=2
     #[inline]
-    pub fn is_hup(&self) -> bool {
-        sys::event::is_hup(&self.inner)
+    pub fn is_read_close(&self) -> bool {
```

Comment:
Should we go with is close or close*d*?

Reply:
@Thomasdezeeuw I had this same thought and thought "close" seemed better because it's the same tense as `is_error` and `is_hup`. Also since these are methods on `Events`, they are used to check if an `Event` is a _close_ event, not if the socket half is _closed_. Do you feel differently about that? I could go either way!

Reply:
I've always preferred readable and writable, reading it as "the event indicates the socket is readable", so it that case closed ("the event indicates the socket is closed") made more sense to me. But I don't really feel strongly either way.

Reply:
I changed the method names as well as constants to be past tense